cli/src/main.rs

```diff
@@ -122,3 +139,17 @@ fn decompile(opts: DecompileOpts) -> Result<(), Error> {
     println!("Output successfully saved to {:?}", opts.output);
     Ok(())
 }
+
+fn lint(opts: LintOpts) -> Result<(), Error> {
+    let sources = std::fs::read_to_string(opts.input)?;
+    let entries = parser::parse(&sources).map_err(|err| Error::SyntaxError(err.to_string()))?;
+
+    match opts.bundle {
+        Some(bundle_path) => {
+            let mut bundle: ScriptBundle = ScriptBundle::load(&mut BufReader::new(File::open(bundle_path)?))?;
+            let mut compiler = Compiler::new(&mut bundle.pool)?;
+            compiler.compile(entries)
+        }
+        None => Ok(())
```

Comment:
hmm does it make sense to run the linter just for a syntax check?
I'd make the bundle path required, because you can't really compile any code without it

Reply:
Wasn't too sure about this, myself. It's mainly about granularity, my reasoning was: without any project setup with path to bundle file, syntax linting is still useful, and can more easily be integrated as editor tool. And yeah, syntax lint doesn't guarantee compilation succeeds, but still catches common errors. Some editors can run commands on save, and can auto-save frequently, running full compilation on every auto-save can become an issue. For those situations, users can configure their editor to run lint without bundle on auto-save, and with bundle on manual save.

Reply:
OK, that's fair enough
I can merge but you've got a README conflict

Reply:
README conflict fixed